C++/syntax_test_c.c

```diff
@@ -1,1555 +1,1250 @@
+//                      ^^^^^^^ constant.numeric.value.c 
+//                             ^ constant.numeric.suffix.c 
+//                              ^^ invalid.illegal.numeric.suffix.c 
 
 /////////////////////////////////////////////
-// Typedefs
+// Operators
 /////////////////////////////////////////////
 
-typedef int myint;
-/* <- keyword.declaration */
-/*          ^ entity.name.type */
-
-typedef struct mystruct {
-/* <- keyword.declaration */
-/*      ^ keyword.declaration.struct.c */
-/*             ^ entity.name.struct.c */
-} mystruct;
-/* ^ entity.name.type.typedef.c */
-
-typedef struct MyStructure {} MyStructure_t;
-/* <- keyword.declaration.type.c */
-/*      ^^^^^^ keyword.declaration.struct.c */
-/*             ^^^^^^^^^^^ entity.name.struct.c */
-/*                         ^ punctuation.section.block.begin.c */
-/*                          ^ punctuation.section.block.end.c */
-/*                            ^^^^^^^^^^^^^ entity.name.type.typedef.c */
+    1 + 2 - 3 * 4 / 5 % 6
+//  ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//    ^ keyword.operator.arithmetic.c 
+//      ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//        ^ keyword.operator.arithmetic.c 
+//          ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//            ^ keyword.operator.arithmetic.c 
+//              ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//                ^ keyword.operator.arithmetic.c 
+//                  ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//                    ^ keyword.operator.arithmetic.c 
+//                      ^ meta.number.integer.decimal.c constant.numeric.value.c 
+    1 | 2 & 3 ^ 4 >> 5 << 6
+//  ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//    ^ keyword.operator.bitwise.c 
+//      ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//        ^ keyword.operator.bitwise.c 
+//          ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//            ^ keyword.operator.bitwise.c 
+//              ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//                ^^ keyword.operator.bitwise.c 
+//                   ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//                     ^^ keyword.operator.bitwise.c 
+//                        ^ meta.number.integer.decimal.c constant.numeric.value.c 
+    1 > 2 >= 3 == 4 != 5 <= 6 < 7
+//  ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//    ^ keyword.operator.comparison.c 
+//      ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//        ^^ keyword.operator.comparison.c 
+//           ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//             ^^ keyword.operator.assignment.c 
+//                ^ meta.number.integer.decimal.c constant.numeric.value.c 
+//                  ^ keyword.operator.arithmetic.c 
+//                   ^ keyword.operator.assignment.c 
```

Comment:
Operator patterns is a sensible part. You need to carefully organize them in correct order, starting with those of 3 chars, followd by 2 and then 1 to ensure they are matched correctly.

```suggestion
//                  ^^ keyword.operator.comparison.c 
```